# アイスクリームの売上

In [71]:
import pandas as pd
import numpy as np

データフレームの設定

In [72]:
df_past = pd.read_csv("sale.csv")
df_future = pd.read_csv("sale_future.csv")

変数の指定

In [73]:
x_name=['temperatuer', 'price', 'rainy']
x = df_past[x_name]
y=df_past["sales"]

ライブラリの導入、回帰分析

In [74]:
import statsmodels.api as sm

model = sm.OLS(y,sm.add_constant(x))
result= model.fit()
result.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                  sales   R-squared:                       0.923
Model:                            OLS   Adj. R-squared:                  0.918
Method:                 Least Squares   F-statistic:                     182.8
Date:                Fri, 22 Apr 2022   Prob (F-statistic):           1.45e-25
Time:                        19:44:31   Log-Likelihood:                -371.07
No. Observations:                  50   AIC:                             750.1
Df Residuals:                      46   BIC:                             757.8
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
===============================================================================
                  coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------------------------
const        -255.0701    243.019     -1.050      0.299    -744.242     234.102
temperatuer   105.1010      4.882     21.528      0.000      95.274     114.928
price          -0.0484      0.721     -0.067      0.947      -1.499       1.402
rainy        -318.5931    139.785     -2.279      0.027    -599.966     -37.220
==============================================================================
Omnibus:                       11.481   Durbin-Watson:                   1.645
Prob(Omnibus):                  0.003   Jarque-Bera (JB):               11.694
Skew:                          -0.972   Prob(JB):                      0.00289
Kurtosis:                       4.354   Cond. No.                     1.26e+03
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 1.26e+03. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

考察

【P>|t|】を見る。0.05より大きい値は帰無仮説を受容するため、関係ない。
したがって気温と天気にのみ売り上げは依存することがわかる。

次に【Adj. R-squared】を見る。これは回帰分析のあてはまりの良さを表す。自由度調整済みR二乗値は0.918であり、1に近いため分析の精度は高いことも分かる。

ここで上記より値段は関係ないことがわかった。次に値段を外した分析を行う。

In [75]:
x_name=['temperatuer', 'rainy']
x = df_past[x_name]
y=df_past["sales"]

In [76]:
model = sm.OLS(y,sm.add_constant(x))
result= model.fit()
result.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                  sales   R-squared:                       0.923
Model:                            OLS   Adj. R-squared:                  0.919
Method:                 Least Squares   F-statistic:                     280.2
Date:                Fri, 22 Apr 2022   Prob (F-statistic):           7.64e-27
Time:                        19:44:32   Log-Likelihood:                -371.08
No. Observations:                  50   AIC:                             748.2
Df Residuals:                      47   BIC:                             753.9
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
===============================================================================
                  coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------------------------
const        -268.2018    142.747     -1.879      0.066    -555.371      18.967
temperatuer   105.0677      4.805     21.865      0.000      95.401     114.735
rainy        -319.7075    137.319     -2.328      0.024    -595.958     -43.457
==============================================================================
Omnibus:                       11.580   Durbin-Watson:                   1.651
Prob(Omnibus):                  0.003   Jarque-Bera (JB):               11.857
Skew:                          -0.975   Prob(JB):                      0.00266
Kurtosis:                       4.376   Cond. No.                         78.3
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

自由度調整済みR二乗値は0.918より0.001上がっており、P値も0.05以内で収まっている。

次に未来の予想を行う。

未来予測

In [77]:
result.predict(sm.add_constant(df_future[x_name]))

0     672.903485
1    1623.017424
2    3409.168950
3     988.106695
4    1513.445380
dtype: float64